x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/DateFieldScript.java

```diff
@@ -75,4 +75,24 @@ public long parse(Object str) {
             return script.formatter.parseMillis(str.toString());
         }
     }
+
+    public static class EmitValues {
+        private final DateFieldScript script;
+
+        public EmitValues(DateFieldScript script) {
+            this.script = script;
+        }
+
+        public void emitFromPath(String path) {
+            for (Object v : script.extractFromSource(path)) {
+                if (v instanceof String) {
+                    try {
+                        script.emit(script.formatter.parseMillis((String) v));
```

Comment:
does this mean that we can remove the parse function from the date context @nik9000 ?

Reply:
I believe we can, yes. We should do that either in this PR or in a follow up right after. Though I think maybe we're better off doing it in this one just because of all the lovely test coverage that adds. It might catch something funny about this PR that we didn't see.

Reply:
++ to doing that here then, it seems like parse won't have any use after this change anyways